yajsapi/utils/log.ts

```diff
@@ -135,15 +137,22 @@ class SummaryLogger {
   }
 
   _print_total_cost(): void {
-    if (!this.finished) return;
-    const provider_names = new Set(Object.keys(this.provider_tasks));
-    if (isSuperset(new Set(Object.keys(this.provider_cost)), provider_names)) {
-      const total_cost = Object.values(this.provider_cost).reduce(
-        (item, acc) => (acc += item),
-        0
-      );
-      logger.info(`Total cost: ${total_cost}`);
-    }
+    const results = [...this.confirmed_agreements].map(
+      (agr_id) => {
+        const name = this.agreement_provider_name[agr_id];
+        const tasks = this.provider_tasks[name];
+        const cost = this.provider_cost[name] || "0 (no invoices?)";
+        return {
+          'Provider Name': name,
+          'Tasks Computed': tasks ? tasks.length : 0,
+          'Total Cost': cost,
+        };
+      }
+    );
+    const total_cost = Object.values(this.provider_cost).reduce((item, acc) => (acc + item), 0);
```

Comment:
One little correction, seems we missed, `accumulator` for the reduce should be on the left side. Current one works cuz anyway we return result of sum, but conceptually it would be better to keep in the right position.
```js
    const total_cost = Object.values(this.provider_cost).reduce((acc, item) => (acc + item), 0);
```